tests/integration/test_integration_all_rust.py

```diff
@@ -214,23 +220,22 @@ def __getattribute__(self, name: str):
         else:
             return f
 
-    def connect(self, connection_port: Optional[int] = None):
+    def connect(self, connection_port: int | None = None):
         url = self.url
         if connection_port:  # pragma: nocover
             url = "ws://localhost:{}/".format(connection_port)
         self.ws = websocket.create_connection(url, header=self.headers)
         return self.ws.connected
 
-    def hello(
-        self, uaid: Optional[str] = None, services: Optional[list[str]] = None
-    ):
+    def hello(self, uaid: str | None = None, services: list[str] | None = None):
+        if not self.ws:
+            raise Exception("WebSocket client not available as expected")
```

Comment:
Is this check necessary? I ask because there's a number of other methods using `self.ws` that could probably also use it, if so -- then it's probably worth a refactor into an `ensure_ws` method.

Reply:
So I added the check to appease mypy, because "technically" self.ws can evaluate to None and we operate with an assumption that it won't be None.

I like the ensure_ws idea. Like this right?
<img width="877" alt="image" src="https://github.com/mozilla-services/autopush-rs/assets/3422688/76820af1-085d-400f-864a-d74720ab859f">
Unfortunately mypy doesn't accept this though
<img width="1062" alt="image" src="https://github.com/mozilla-services/autopush-rs/assets/3422688/069b7784-266e-444c-b2f8-c433cb188788">

Reply:
Ok, totally disregard my comment. This is just silly mypy implementation details that aren't worth bothering with for now